app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/login/SignUpFragment.java

```diff
@@ -70,15 +75,17 @@ private void editTextAndInputLayoutDeclaration() {
     /**
      * This method is responsible for the creation and validation of the new user.
      */
-    private void userCreationAndValidation() {
+    private void userCreationAndValidation() throws ExecutionException, InterruptedException {
+        userManagerService.usernameExists(username);
```

Comment:
Here should be an if statement to check whether the username already exists or not